File: apps/web/types/lesson.ts
//apps/web/types/lesson.ts

export type LessonSection = {
    title: string;
    content: string;
};

export type LessonContent = {
    title: string;
    today_focus: string;
    summary: string;
    sections: LessonSection[];
    next_step: string;
};

export type LessonResource = {
    title: string;
    url: string;
    type: string;
    reason: string;
    snippet?: string | null;
};

export type LessonData = {
    topic: string;
    level: string;
    roadmap: string[];
    lesson: LessonContent;
    resources: LessonResource[];
};

export type SavedLesson = LessonData & {
    id: string;
};

export type LessonPreview = {
    id: string;
    topic: string;
    level: string;
    title: string;
};

// export type LessonSection = {
//     title: string;
//     content: string;
// };

// export type LessonContent = {
//     title: string;
//     today_focus: string;
//     summary: string;
//     sections: LessonSection[];
//     next_step: string;
// };

// export type LessonResource = {
//     title: string;
//     url: string;
//     type: string;
//     reason: string;
//     snippet?: string | null;
// };

// export type SavedLesson = {
//     id: string;
//     topic: string;
//     level: string;
//     roadmap: string[];
//     lesson: LessonContent;
//     resources: LessonResource[];
// };

// export type LessonPreview = {
//     id: string;
//     topic: string;
//     level: string;
//     title: string;
// };

File: apps/api/app/schemas/lesson.py
#apps/api/app/schemas/lesson.py

from pydantic import BaseModel, Field
from typing import List


class LessonSection(BaseModel):
    title: str
    content: str


class LessonContent(BaseModel):
    title: str
    today_focus: str
    summary: str
    sections: List[LessonSection]
    next_step: str


class LessonResource(BaseModel):
    title: str
    url: str
    type: str
    reason: str
    snippet: str | None = None


class LessonDeepDiveItem(BaseModel):
    title: str
    url: str | None = None
    type: str
    reason: str
    snippet: str | None = None


class GenerateLessonRequest(BaseModel):
    topic: str = Field(..., min_length=1)
    level: str = Field(..., min_length=1)


class GeneratedLessonResponse(BaseModel):
    topic: str
    level: str
    roadmap: List[str]
    lesson: LessonContent
    resources: List[LessonResource]
    deepDive: List[LessonDeepDiveItem] = Field(default_factory=list)


class SaveLessonRequest(BaseModel):
    topic: str = Field(..., min_length=1)
    level: str = Field(..., min_length=1)
    roadmap: List[str]
    lesson: LessonContent
    resources: List[LessonResource]
    deepDive: List[LessonDeepDiveItem] = Field(default_factory=list)


class LessonResponse(BaseModel):
    id: str
    topic: str
    level: str
    roadmap: List[str]
    lesson: LessonContent
    resources: List[LessonResource]
    deepDive: List[LessonDeepDiveItem] = Field(default_factory=list)